applications/TopologyOptimizationApplication/custom_strategies/structure_adjoint_sensitivity_strategy.h

```diff
@@ -0,0 +1,158 @@
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Baumgärtner Daniel, https://github.com/dbaumgaertner
+//                   Octaviano Malfavón Farías
+//                   Eric Gonzales
+//					 Philipp Hofer
+//					 Erich Wehrle
+//
+// ==============================================================================
+
+#if !defined(KRATOS_STRUCTURE_ADJOINT_SENSITIVITY_STRATEGY_H_INCLUDED)
+#define  KRATOS_STRUCTURE_ADJOINT_SENSITIVITY_STRATEGY_H_INCLUDED
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+#include "solving_strategies/strategies/solving_strategy.h"
+#include "custom_elements/small_displacement_simp_element.h"
+
+// Application includes
+#include "topology_optimization_application.h"
+
+
+namespace Kratos {
+
+///@addtogroup TopologyOptimizationApplication
+///@{
+
+///@name Kratos Classes
+///@{
+
+/// Solution strategy to calculate the sensitivities.
+/// Derives from the previously defined Solving Strategy
+
+template    <class TSparseSpace, 
+            class TDenseSpace, 
+            class TLinearSolver
+            >
+class StructureAdjointSensitivityStrategy
+    : public SolvingStrategy<TSparseSpace, TDenseSpace>
+{
+public:
+
+    ///@name Type Definitions
+    ///@{
+
+    KRATOS_CLASS_POINTER_DEFINITION(StructureAdjointSensitivityStrategy);
+
+    typedef SolvingStrategy<TSparseSpace,TDenseSpace> BaseType;
+
+    typedef typename Scheme<TSparseSpace,TDenseSpace>::Pointer SchemePointerType;
+    typedef typename BuilderAndSolver<TSparseSpace,TDenseSpace,TLinearSolver>::Pointer BuilderAndSolverPointerType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    StructureAdjointSensitivityStrategy( ModelPart& rStructureModelPart,
+            typename TLinearSolver::Pointer pNewLinearSolver,
+            const int dimension = 3)
+    : BaseType(rStructureModelPart),
+        mr_structure_model_part(rStructureModelPart),
+        m_dimension(dimension)
+    {}
+
+
+    ///virtual ~StructureAdjointSensitivityStrategy()
+    ~StructureAdjointSensitivityStrategy()	override
+    {}
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    // ---------------------------------------------------------------------------------------------------------------------------------------------
+    // --------------------------------- COMPUTE SENSITIVITIES  ------------------------------------------------------------------------------------
+    // ---------------------------------------------------------------------------------------------------------------------------------------------
+
+    /// Computes DCDX sensitivities from the adjoint solution
+    void ComputeStrainEnergySensitivities()
+    {
+        KRATOS_TRY;
+
+        double Out = 0.0;
+        int i= 0;
+
+        clock_t begin = clock();
+
+        for ( ModelPart::ElementIterator element_i = mr_structure_model_part.ElementsBegin(); element_i!= mr_structure_model_part.ElementsEnd();
```

Comment:
Do it as parallel loop

Reply:
use the pararell utilities please

Reply:
This is still not using parallel utilities

Reply:
Is using `#pragma omp parallel for` not the right choice to use the parallel utility?

Reply:
Using this: https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/utilities/parallel_utilities.h

Example: https://github.com/KratosMultiphysics/Kratos/blob/f55a2c57ac8102038bdb46428cb2d2a95c7b92cc/kratos/utilities/auxiliar_model_part_utilities.h#L308